src/main/java/org/mskcc/cmo/metadb/service/impl/ValidRequestCheckerImpl.java

```diff
@@ -0,0 +1,268 @@
+package org.mskcc.cmo.metadb.service.impl;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import org.apache.commons.lang3.EnumUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.logging.log4j.util.Strings;
+import org.mskcc.cmo.common.enums.CmoSampleClass;
+import org.mskcc.cmo.common.enums.SampleOrigin;
+import org.mskcc.cmo.common.enums.SampleType;
+import org.mskcc.cmo.common.enums.SpecimenType;
+import org.mskcc.cmo.metadb.service.ValidRequestChecker;
+import org.mskcc.cmo.metadb.service.util.RequestStatusLogger;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+@Component
+public class ValidRequestCheckerImpl implements ValidRequestChecker {
+
+    private final ObjectMapper mapper = new ObjectMapper();
+    private boolean isCmoRequest;
+    private boolean hasRequestId;
+
+    @Autowired
+    private RequestStatusLogger requestStatusLogger;
+
+    @Value("${igo.cmo_request_filter:false}")
+    private Boolean igoCmoRequestFilter;
+
+    private static final Log LOG = LogFactory.getLog(ValidRequestCheckerImpl.class);
+
+    /**
+     * Checks is the request is a cmoRequest and has a requestId
+     * Sample list is extracted from the requestJsonMap
+     * If the total number of samples available is equal to valid samples,
+     * then the request as a whole is valid
+     * If the number of valid samples is zero, the request is invalid and will be logged
+     * If the number of valid samples is less than the total number of samples, the
+     * request is still considered valid but logged to keep note of the invalid samples
```

Comment:
Maybe make a bullet point list of these things for the javadoc? That way it's more readable